### DISCUSSION TITLE
[Prometheus Plugin] Feature Request: Being Able To Configure Exposed Labels On Metrics

### DISCUSSION THREAD

**Discussion by mnofresno** [Kong/kong#9116] [Ideas and feature requests]
In my team we monitor our APIs using Kong with Prometheus plugin and use an alert manager label called "service" used by the whole company in order to create routing rules for the alerting emails.

This label collides with Kong's metric 'service' label throwing this warn on prometheus:

`err="vector contains metrics with the same labelset after applying alert labels"`

After a while researching I've noticed we could be able to configure the metrics data labels on the plugin just adding a simple configuration option on the plugin's schema and using it on kong/plugins/prometheus/prometheus.lua

**Comment by mnofresno** [comment]
In order to avoid breaking backwards compatibility the config should default to current hardcoded label names.

I think the default labels used in this plugin: 'service', 'instance', etc.. are pretty error prone because them would probably collide with some other labels configured on any big architecture.

**Reply by sydpz** [reply]
similar problem, how do you solve it?
